aten/src/ATen/native/cpu/UnaryOpsKernel.cpp

```diff
@@ -440,24 +440,60 @@ void normal_fill(Tensor& self, const scalar_t mean, const scalar_t std, Generato
   }
 }
 
+std::vector<int64_t> computeStrideForComplex(IntArrayRef oldstride) {
+  auto res = oldstride.vec();
+  for(size_t i = 0; i < res.size(); i++) {
+    res[i] = res[i] * 2;
+  }
+  res.emplace_back(1);
+  return res;
+}
+
+Tensor view_complex_as_float(const Tensor& self) {
+  auto new_sizes = self.sizes().vec();
+  // last dimension will always have two elements containing the real and imag vals
+  new_sizes.emplace_back(2);
+  auto new_strides = computeStrideForComplex(self.strides());
```

Comment:
Good! It looks like this should work for an arbitrary view even if it's not contiguous. It would be cool to test this; I bet if you run `normal_` on a non-contiguous tensor you should be able to see if it works or not.

Reply:
yeah it seems to work for non-contiguous tensors. tried this:
```
>>>x=torch.zeros(2,3,4, dtype = torch.complex64)
>>>x.transpose_()
>>> x.normal_()
tensor([[[                         (0.6367 - 0.4202j),
                                  (-0.2379 + 0.1448j),
                                   (0.0171 + 0.3797j)],
         [                         (0.4977 + 0.4061j),
                                   (0.8147 - 0.7232j),
                                   (0.8197 + 0.4234j)],
         [                         (1.3404 - 0.1096j),
                                   (0.3512 - 0.9325j),
                                   (0.2970 - 0.6507j)],
         [                        (-1.1207 - 0.6621j),
                                   (0.5605 - 0.0454j),
                                  (-0.2007 + 0.8882j)]],

        [[                         (0.4621 + 0.2492j),
                                  (-0.2410 + 1.0988j),
                                  (-0.0941 - 0.0518j)],
         [                         (0.7760 + 0.7900j),
                                   (0.4653 - 0.3250j),
                                   (0.2949 - 0.6387j)],
         [                        (-0.3772 - 0.2726j),
                                  (-1.4134 + 0.2974j),
                                   (1.6450 + 0.4730j)],
         [                         (0.6167 - 0.8038j),
                                   (0.0074 - 0.4632j),
                                  (-0.3093 - 0.2770j)]]],
       dtype=torch.complex64)

```

Reply:
but I think it's worth adding a test for it. thanks!